xclim/sdba/properties.py

```diff
@@ -403,112 +406,186 @@ def _acf(ds, *, dim, lag, freq):
-    amp.attrs["units"] = xc.core.units.ensure_delta(units)
-    if amplitude_type == "relative":
-        amp = amp * 100 / da.mean(dim=group.dim, keep_attrs=True)
-    amp = amp.mean(dim=group.dim)
-    return amp.assign_attrs(units="%" if amplitude_type == "relative" else units)
+
+    ac = da.groupby("time.dayofyear").mean()
+    if window > 1:  # smooth the cycle
+        # We want the rolling mean to be circular. There's no built-in method to do this in xarray,
+        # we'll pad the array and extract the meaningful part.
+        ac = (
+            ac.pad(dayofyear=(window // 2), mode="wrap")
+            .rolling(dayofyear=window, center=True)
+            .mean()
+            .isel(dayofyear=slice(window // 2, -(window // 2)))
+        )
+    # TODO: In April 2024, use a match-case.
+    if stat == "absamp":
+        out = ac.max("dayofyear") - ac.min("dayofyear")
+        out.attrs["units"] = xc.core.units.ensure_delta(units)
+    elif stat == "relamp":
+        out = (ac.max("dayofyear") - ac.min("dayofyear")) * 100 / ac.mean("dayofyear")
+        out.attrs["units"] = "%"
+    elif stat == "phase":
+        out = ac.idxmax("dayofyear")
+        out.attrs.update(units="", is_dayofyear=np.int32(1))
+    elif stat == "min":
+        out = ac.min("dayofyear")
+        out.attrs["units"] = units
+    elif stat == "max":
+        out = ac.max("dayofyear")
+        out.attrs["units"] = units
+    elif stat == "asymmetry":
+        out = (ac.idxmax("dayofyear") - ac.idxmin("dayofyear")) % 365 / 365
+        out.attrs["units"] = "yr"
+    else:
+        raise NotImplementedError(f"{stat} is not a valid annual cycle statistic.")
+    return out
 
 
 annual_cycle_amplitude = StatisticalProperty(
     identifier="annual_cycle_amplitude",
     aspect="temporal",
-    compute=_annual_cycle_amplitude,
-    parameters={"amplitude_type": "absolute"},
+    compute=_annual_cycle,
+    parameters={"stat": "absamp"},
     allowed_groups=["group"],
     cell_methods="time: range time: mean",
 )
 
-
 relative_annual_cycle_amplitude = StatisticalProperty(
     identifier="relative_annual_cycle_amplitude",
     aspect="temporal",
-    compute=_annual_cycle_amplitude,
-    parameters={"amplitude_type": "relative"},
+    compute=_annual_cycle,
+    units="%",
+    parameters={"stat": "relamp"},
     allowed_groups=["group"],
     cell_methods="time: range time: mean",
     measure="xclim.sdba.measures.RATIO",
 )
 
+annual_cycle_phase = StatisticalProperty(
+    identifier="annual_cycle_phase",
+    aspect="temporal",
+    units="",
+    compute=_annual_cycle,
+    parameters={"stat": "phase"},
+    cell_methods="time: range",
+    allowed_groups=["group"],
+    measure="xclim.sdba.measures.CIRCULAR_BIAS",
+)
+
+annual_cycle_asymmetry = StatisticalProperty(
+    identifier="annual_cycle_asymmetry",
+    aspect="temporal",
+    compute=_annual_cycle,
+    parameters={"stat": "asymmetry"},
+    allowed_groups=["group"],
+    units="yr",
+)
```

Comment:
min and max exist in stat, but they are not create as indicators. Is this on purpose ?

Reply:
More like laziness. I'll add them :).